src/Compilers/CSharp/Portable/Binder/PatternVariableFinder.cs

```diff
@@ -165,6 +165,21 @@ public override void VisitArgument(ArgumentSyntax node)
                 return;
             }
 
+            switch (node.Identifier.
+                         Parent. // ArgumentSyntax
+                         Parent. // ArgumentListSyntax
+                         Parent. // invocation/constructor initializer
+                         Kind())
+            {
+                case SyntaxKind.InvocationExpression:
+                case SyntaxKind.ObjectCreationExpression:
+                case SyntaxKind.ThisConstructorInitializer:
+                case SyntaxKind.BaseConstructorInitializer:
+                    break;
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(node.Identifier.Parent.Parent.Parent.Kind());
```

Comment:
I looked through other places where we throw UnexpectedValue, not a single one of them uses more detailed message. I will add a comment instead.

---
In reply to: [64272964](https://github.com/dotnet/roslyn/pull/11493#discussion_r64272964) [](ancestors = 64272964)